src/runtime/importhook.cs

```diff
@@ -107,29 +107,30 @@ internal static void Shutdown()
             }
 
             RestoreImport();
-
             bool shouldFreeDef = Runtime.Refcount(py_clr_module) == 1;
-            Runtime.XDecref(py_clr_module);
-            py_clr_module = IntPtr.Zero;
-            if (shouldFreeDef)
+#if !NETSTANDARD
+            if(Runtime.ShutdownMode != ShutdownMode.Reload)
+#endif
```

Comment:
3.0 is supposed to only be .NET Standard, so I think this is incorrect. I suspect this should instead be a runtime check if `AppDomain` is fully supported.

Reply:
I don't think AppDomain needs to be supported. I've not done it but if the domain is started/stopped by the mono runtime at C/C++ level that counts too, no?